packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -186,11 +206,11 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
    * @param opts - Additional options
    */
   async replace(content: T | null, opts: UpdateOpts = {}): Promise<void> {
-    opts = { ...DEFAULT_UPDATE_OPTS, ...opts }
+    const { header, ...options } = { ...DEFAULT_UPDATE_OPTS, ...opts }
```

Comment:
Can we add an arg to this function for any metadata updates, rather than jamming the header into the `opts` (which is violating the type definition of `UpdateOpts`)?

Reply:
Yeah, that is a bit of a bummer.  Ideally we would insert the `metadata` arg between `content` and `opts`, but that would be a breaking change to the API.  Maybe we should consider a major version bump for the ModelInstanceDocument package?  I really don't like putting data into the `UpdateOpts`, that is fundamentally changing the concept of the `UpdateOpts` type.  At that point, we might as well just have a single argument to all these functions that is a dictionary of all the functions arguments, and never use multiple arguments for anything.